### PR TITLE
feat: add missing columns in information_schema.columns and add current_setting func

### LINKED ISSUE #10051: Support `current_setting`
### Describe the bug

```
2023-05-29T11:40:41.350829Z ERROR risingwave_frontend::session: failed to handle sql:
SELECT CAST(current_setting('server_version_num') AS INT) / 100 AS version:
    Bind error: failed to bind expression: CAST(current_setting('server_version_num') AS INT) / 100
    Caused by:
      Feature is not yet implemented: unsupported function: "current_setting"
      Tracking issue: https://github.com/risingwavelabs/risingwave/issues/112
```
when connecting from Grafana by creating `RW` as a PostgreSQL data source.

See #8345

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

Don't know if we should customize a plugin of Risingwave for Grafana though

### LINKED ISSUE #10131: tracking: support for golang ORM library 'xorm'
### Is your feature request related to a problem? Please describe.

XORM: https://xorm.io/

- [x] pg_tables https://www.postgresql.org/docs/current/view-pg-tables.html
    - [x] tablename
    - [x] schemaname
- [x] INFORMATION_SCHEMA.COLUMNS https://www.postgresql.org/docs/current/infoschema-columns.html 
    - [x] column_default
    - [x] character_maximum_length
- [x] CHAR
The actual query:

```sql
SELECT column_name, column_default, is_nullable, data_type, character_maximum_length, description,
    CASE WHEN p.contype = 'p' THEN true ELSE false END AS primarykey,
    CASE WHEN p.contype = 'u' THEN true ELSE false END AS uniquekey
FROM pg_attribute f
    JOIN pg_class c ON c.oid = f.attrelid JOIN pg_type t ON t.oid = f.atttypid
    LEFT JOIN pg_attrdef d ON d.adrelid = c.oid AND d.adnum = f.attnum
    LEFT JOIN pg_description de ON f.attrelid=de.objoid AND f.attnum=de.objsubid
    LEFT JOIN pg_namespace n ON n.oid = c.relnamespace
    LEFT JOIN pg_constraint p ON p.conrelid = c.oid AND f.attnum = ANY (p.conkey)
    LEFT JOIN pg_class AS g ON p.confrelid = g.oid
    LEFT JOIN INFORMATION_SCHEMA.COLUMNS s ON s.column_name=f.attname AND c.relname=s.table_name
WHERE n.nspname= s.table_schema AND c.relkind = 'r'::char AND c.relname = 'a' AND f.attnum > 0 ORDER BY f.attnum;
```

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

To support golang ORM library 'xorm' and for grafana, this PR adds some missing columns in `information_schema.columns`, add a `current_setting` function and fixes several issues.
Related #10131 , #10134 .  Resolve #10051 .

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- SQL commands, functions, and operators

### Release note

</details>

1. add some missing columns in `information_schema.columns`: column_default, character_maximum_length and udt_name.
2. add a new sql function: `current_setting`.

### CODE REVIEW

File: src/frontend/src/binder/expr/mod.rs [hunk 1]
```diff
@@ -565,13 +565,8 @@ pub fn bind_data_type(data_type: &AstDataType) -> Result<DataType> {
...
         AstDataType::Array(datatype) => DataType::List(Box::new(bind_data_type(datatype)?)),
-        AstDataType::Char(..) => {
-            return Err(ErrorCode::NotImplemented(
-                "CHAR is not supported, please use VARCHAR instead\n".to_string(),
-                None.into(),
-            )
-            .into())
-        }
+        // workaround to bind char type to varchar.
+        AstDataType::Char(..) => DataType::Varchar,
```
Comment (xiangjinwu) [inline src/frontend/src/binder/expr/mod.rs:569]: `char(n)` has different semantic from (unlimited) `varchar`. People creating `char(n)` columns would get wrong results when we process them as `varchar`.

File: src/frontend/src/binder/expr/function.rs [hunk 2]
```diff
@@ -743,6 +743,25 @@ impl Binder {
...
+                        )
+                        .into());
+                    };
+                    let Some(ScalarImpl::Utf8(input)) = literal.get_data() else {
+                        return Err(ErrorCode::ExprError(
+                            "Only string literal is supported in `current_setting`.".into(),
+                        )
+                        .into());
+                    };
+                    match binder.session_config.get(&input.to_string()) {
```
Comment (xiangjinwu) [inline src/frontend/src/binder/expr/function.rs:760]:
```suggestion
                    match binder.session_config.get(input.as_ref()) {
```

**Review by wangrunji0408 — COMMENTED** [review]
LGTM

File: src/frontend/src/binder/expr/mod.rs (hunk 1 above)
Comment (yezizp2012) [reply src/frontend/src/binder/expr/mod.rs]: Indeed, let's leave it unchanged and try to resolve it in the upstream side.

**Review by xiangjinwu — APPROVED** [review]

**Pull request merged**